src/test/java/org/jenkins/plugins/lockableresources/LockStepTest.java

```diff
@@ -976,4 +976,35 @@ public void skipIfLocked() throws Exception {
     j.assertLogContains("[resource1] is locked, skipping execution...", b1);
     j.assertLogNotContains("Running body", b1);
   }
+
+  @Test
+  public void lockWithoutUsingBody() throws Exception {
+    LockableResourcesManager lm = LockableResourcesManager.get();
+    lm.createResourceWithLabel("resource1", "label1");
+    lm.reserve(Arrays.asList(lm.fromName("resource1")), "test");
+
+    WorkflowJob p = j.jenkins.createProject(WorkflowJob.class, "p");
+    p.setDefinition(
+        new CpsFlowDefinition(
+            "def lockObject = lock(resource: 'resource1')\n"
+                + "try  {\n"
+                + "  println \"Locked: \" + lockObject\n"
+                + "} finally {\n"
+                + "  lockObject.release()\n"
+                + "}\n"
+                + "println \"Lock is now released\"\n"
+                + "def lockObject2 = lock(resource: 'resource1')\n"
+                + "try  {\n"
+                + "  println \"Locked2: \" + lockObject2\n"
+                + "} finally {\n"
+                + "  lockObject2.release()\n"
+                + "}\n"));
```

Comment:
```suggestion
                + "}\n", true));
```

which would show you that you needed to add `@Whitelisted` to the `release` method. **But**

> a lock object that has a `release()`-method

As a general rule, this is a bad idea in my experience: while Pipeline steps may return value objects for various purposes, these objects must not have any mutating methods. If you want to release a lock, define an explicit release step.

Reply:
That would be ok with me. That was in my first proposal https://github.com/jenkinsci/lockable-resources-plugin/pull/179

I can change it back again, what do you say @TobiX ?

Reply:
I guess I missed https://github.com/jenkinsci/lockable-resources-plugin/pull/179#issuecomment-590089347 (have not been following all this very closely).

Reply:
Ah. A pity, I thought we could avoid a new release step. I'll trust @jglick on this, so maybe we should just go with the original proposal then: A pair of new steps:
- `acquireLock` returning an (immutable?, serializable) value-object
- `releaseLock` accepting this value-object to release the lock again
I would probably mark those steps as "advanced" (Implement `StepDescriptor.isAdvanced()` to return `true`), so they appear in the "advanced" section of the snippet generator...

Reply:
Any progress on this PR?
We have pipeline to deploy a full system. The process is very heavy which takes hours.  If the system is not stable （check by tests）, we need to keep it for investigation. So I'd like to lock the resources to prevent a new deployment by accident. So, how can I lock the resource in pipeline without auto release?

Btw, it's better that I can force release a locked resource in another cases.

Reply:
@topikachu Your usecase is not covered by this pull request - And I'm very hesitant to add something like this. Locks made by this plugin currently only persist until the end of a job. If you need a different behavior, please discuss on the already existing ticket #120 or provide a pull request...

Reply:
> @topikachu Your usecase is not covered by this pull request

Is it not covered by being able to lock those resources like this?

```
parallel(
  a: {
      lockObject = lock(resource: 'staging-server')
  },
  b: {
      lockObject2 = lock(resource: 'staging-server2')
  }
)

```
They would be auto released when the job is done but he can keep the job running.

Reply:
Ah right: That even sounds like the right solution when not using this plugin at all: For example, with on-demand agents (Kubernetes etc.) you could keep the agent alive until someone could debug the error state and only then release all resources associated with it... (Gives me another argument to keep my stance on always releasing locks at the end of a build 😛)

Reply:
Any updates on this PR? The solution outlined should give us a path forward in being able to lock concurrent runs of a multibranch pipline, so looking forward to having this functionality in the plugin!

Reply:
I gave up because of poor communication with maintainers.

Reply:
@tomasbjerre I'm very sorry if you see it that way. My most recent suggestion (2020-04-13) was going back to the original proposal, since @jglick suggested against using mutable objects inside pipeline. Was there anything unclear about that?

Reply:
No that part is clear.